app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt

```diff
@@ -167,76 +178,105 @@ class ConversationListViewModel @Inject constructor(
+                                       selfTeamId: TeamId?,
+                                       userTypeMapper: UserTypeMapper): ConversationItem =
+    when (this) {
+        is Group -> {
+            ConversationItem.GroupConversation(
+                groupName = conversation.name.orEmpty(),
+                conversationId = conversation.id,
+                mutedStatus = conversation.mutedStatus,
+                isLegalHold = legalHoldStatus.showLegalHoldIndicator(),
+                lastEvent = ConversationLastEvent.None, // TODO implement unread events
+                hasOnGoingCall = hasOngoingCall
+            )
+        }
+        is OneOne -> {
+            ConversationItem.PrivateConversation(
+                userAvatarData = UserAvatarData(
+                    otherUser.previewPicture?.let { UserAvatarAsset(wireSessionImageLoader, it) },
+                    UserAvailabilityStatus.NONE // TODO Get actual status
+                ),
+                conversationInfo = ConversationInfo(
+                    name = otherUser.name.orEmpty(),
+                    membership = userTypeMapper.toMembership(userType)
+                ),
+                conversationId = conversation.id,
+                mutedStatus = conversation.mutedStatus,
+                isLegalHold = legalHoldStatus.showLegalHoldIndicator(),
+                lastEvent = ConversationLastEvent.None, // TODO implement unread events
+                userId = otherUser.id,
+                blockingState = otherUser.getBlockingState(selfTeamId)
+            )
+        }
+        is Connection -> {
+            ConversationItem.ConnectionConversation(
+                userAvatarData = UserAvatarData(
+                    otherUser?.previewPicture?.let { UserAvatarAsset(wireSessionImageLoader, it) },
+                    UserAvailabilityStatus.NONE // TODO Get actual status
+                ),
+                conversationInfo = ConversationInfo(
+                    name = otherUser?.name.orEmpty(),
+                    membership = userTypeMapper.toMembership(userType)
+                ),
+                lastEvent = ConversationLastEvent.Connection(
+                    connection.status,
+                    connection.qualifiedToId
+                ),
+                conversationId = conversation.id,
+                mutedStatus = conversation.mutedStatus,
+                connectionState = connection.status
+            )
+        }
+        is Self -> {
+            throw IllegalArgumentException("Self conversations should not be visible to the user.")
+        }
+        else -> {
+            throw IllegalArgumentException("$this conversations should not be visible to the user.")
+        }
+    }
+
+private fun OtherUser.getBlockingState(selfTeamId: TeamId?): BlockingState =
+    when {
+        connectionStatus == ConnectionState.BLOCKED -> BlockingState.BLOCKED
+        teamId == selfTeamId -> BlockingState.CAN_NOT_BE_BLOCKED
```

Comment:
Where is being checked that the blockUserUseCase can't be invoked for users of same team?

Reply:
that is one of the requirements in task: user can block only users that are not in his/her team